diagnostic_feedback/quiz.py

```diff
@@ -0,0 +1,436 @@
+    )
+
+    student_choices = Dict(
+        default={},
+        help=_("This will hold user provided answers of questions"),
+        scope=Scope.user_state,
+    )
+
+    quiz_type = String(
+        default="",
+        scope=Scope.content,
+        help=_("Type of quiz")
+    )
+
+    results = List(
+        default=[],
+        scope=Scope.content,
+        help=_("List of results")
+    )
+
+    student_result = String(
+        default='',
+        scope=Scope.user_state,
+        help=_("Calculated feedback of each user")
+    )
+
+    types = List(
+        default=[
+            {"value": BUZZFEED_QUIZ_VALUE, "label": BUZZFEED_QUIZ_LABEL},
+            {"value": DIAGNOSTIC_QUIZ_VALUE, "label": DIAGNOSTIC_QUIZ_LABEL},
+        ],
+        scope=Scope.content,
+        help=_("List of results")
+    )
+
+    groups = List(
+        default=[DEFAULT_GROUP],
+        scope=Scope.content,
+        help=_("List of results")
+    )
+
+    current_step = Integer(
+        default=0,
+        scope=Scope.user_state,
+        help=_("To control which question should be shown to student")
+    )
+
+    weight = Float(
+        display_name=_("Weight"),
+        help=_("Defines the maximum total grade of this question."),
+        default=1,
+        scope=Scope.content,
+        enforce_type=True
+    )
+
+    completed = Boolean(
+        default=False,
+        scope=Scope.user_state,
+        help=_("Has the student completed this quiz")
+    )
+
+    @property
+    def display_name_with_default(self):
+        return self.title
+
+    @property
+    def additional_publish_event_data(self):
+        return {
+            'user_id': self.scope_ids.user_id,
+            'block_id': self.get_block_id(),
+            'component_id': self.scope_ids.usage_id
+        }
+
+    has_score = True
+
+    def get_fragment(self, context, view='studio', json_args=None):
+        """
+        return fragment after loading css/js/html either for studio OR student view
+        :param context: context for templates
+        :param view: view_type i;e studio/student
+        :return: fragment after loading all assets
+        """
+        """
+            Return fragment after adding required css/js/html
+        """
+        fragment = Fragment()
+        self.add_templates(fragment, context, view)
+        self.add_css(fragment, view)
+        self.add_js(fragment, view)
+        self.initialize_js_classes(fragment, view, json_args)
+        return fragment
+
+    def append_choice(self, questions):
+        """
+        append student choice with each question if available
+        :param questions: list of questions
+        :return:
+        """
+        """
+
+        """
+        for question in questions:
+            if self.quiz_type == self.DIAGNOSTIC_QUIZ_VALUE:
+                question['student_choice'] = float(self.student_choices.get(question['id'])) if \
+                    self.student_choices.get(question['id']) else ''
```

Comment:
If the value of `student_choices[question['id']]` is `0`, this will return an empty string.  Should the middle of this line be `...if question['id'] in self.student_choices then ...`?